gradoop-flink/src/main/java/org/gradoop/flink/model/api/operators/BaseGraphCollectionOperatorSupport.java

```diff
@@ -0,0 +1,110 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.flink.model.api.operators;
+
+import org.gradoop.flink.model.api.epgm.BaseGraph;
+import org.gradoop.flink.model.api.epgm.BaseGraphCollection;
+import org.gradoop.flink.model.impl.operators.combination.Combination;
+import org.gradoop.flink.model.impl.operators.exclusion.Exclusion;
+import org.gradoop.flink.model.impl.operators.overlap.Overlap;
+
+/**
+ * Defines callFor methods for graph collections.
+ *
+ * @param <LG> graph type
+ * @param <GC> graph collection type
+ */
+public interface BaseGraphCollectionOperatorSupport<LG extends BaseGraph, GC extends BaseGraphCollection> {
+
+  /**
+   * Creates a value using the given unary collection to value operator.
+   *
+   * @param operator unary graph collection to value operator
+   * @param <T> return type
+   * @return result of given operator
+   */
+  <T> T callForValue(UnaryBaseGraphCollectionToValueOperator<GC, T> operator);
+
+  /**
+   * Calls the given binary collection to value operator using that graph collection and the
+   * input graph collection.
+   *
+   * @param operator        binary collection to value operator
+   * @param otherCollection second input collection for operator
+   * @param <T> return type
+   * @return result of given operator
+   */
+  <T> T callForValue(BinaryBaseGraphCollectionToValueOperator<GC, T> operator, GC otherCollection);
+
+  /**
+   * Creates a graph collection using the given unary graph collection operator.
+   *
+   * @param operator unary graph collection to graph collection operator
+   * @return result of given operator
+   */
+  default GC callForCollection(UnaryBaseGraphCollectionToBaseGraphCollectionOperator<GC> operator) {
+    return callForValue(operator);
+  }
+
+  /**
+   * Calls the given binary collection to collection operator using that graph collection and the
+   * input graph collection.
+   *
+   * @param operator        binary collection to collection operator
+   * @param otherCollection second input collection for operator
+   * @return result of given operator
+   */
+  default GC callForCollection(BinaryBaseGraphCollectionToBaseGraphCollectionOperator<GC> operator,
+                               GC otherCollection) {
+    return callForValue(operator, otherCollection);
+  }
+
+  /**
+   * Applies a given unary graph to graph operator (e.g., aggregate) on each
+   * base graph in the graph collection.
+   *
+   * @param operator applicable unary graph to graph operator
+   * @return collection with resulting logical graphs
+   */
+  default GC apply(ApplicableUnaryBaseGraphToBaseGraphOperator<GC> operator) {
+    return callForValue(operator);
+  }
+
+  /**
+   * Transforms a graph collection into a base graph by applying a
```

Comment:
Base graph is not a an actual type, just a base for all types, I think it should therefore just be `graph`
```suggestion
   * Transforms a graph collection into a graph by applying a
```